lib/utils.js

```diff
@@ -68,7 +68,9 @@ exports.TRANSIENT_HTTP_ERROR_CODES = TRANSIENT_HTTP_ERROR_CODES;
 var retry = function retry(f, options) {
   var retry = 0;
   function attempt() {
-    return f(retry).catch(function(err) {
+    return new Promise(function(accept) {
+      accept(f(retry));
+    }).catch(function(err) {
```

Comment:
@jhford, this will cost a tiny bit more... But it should be a lot better at catching synchronous errors, though I expected they were already getting catched.